pkg/handler/v2/alertmanager/alertmanager.go

```diff
@@ -0,0 +1,214 @@
+		if err != nil {
+			return nil, err
+		}
+
+		alertmanager, config, err := alertmanagerProvider.Get(c, userInfo)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+		return convertInternalToAPIAlertmanager(alertmanager, config), nil
+	}
+}
+
+func CreateEndpoint(userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider,
+	privilegedProjectProvider provider.PrivilegedProjectProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(createAlertmanagerReq)
+
+		if err := req.validateCreateAlertmanagerReq(); err != nil {
+			return nil, utilerrors.NewBadRequest(fmt.Errorf("invalid alertmanager configuration: %w", err).Error())
+		}
+
+		c, userInfo, alertmanagerProvider, err := getClusterUserInfoAlertmanagerProvider(ctx, projectProvider, privilegedProjectProvider, userInfoGetter, req.ProjectID, req.ClusterID)
+		if err != nil {
+			return nil, err
+		}
+
+		alertmanager, configSecret := convertAPIToInternalAlertmanager(c, &req.Body)
+		al, config, err := alertmanagerProvider.Create(alertmanager, configSecret, userInfo)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+		return convertInternalToAPIAlertmanager(al, config), nil
+	}
+}
+
+func DeleteEndpoint(userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider,
+	privilegedProjectProvider provider.PrivilegedProjectProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(deleteAlertmanagerReq)
+
+		c, userInfo, alertmanagerProvider, err := getClusterUserInfoAlertmanagerProvider(ctx, projectProvider, privilegedProjectProvider, userInfoGetter, req.ProjectID, req.ClusterID)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := alertmanagerProvider.Delete(c, userInfo); err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+		return nil, nil
+	}
+}
+
+// getAlertmanagerReq defines HTTP request for getting alertmanager
+// swagger:parameters getAlertmanager
+type getAlertmanagerReq struct {
+	cluster.GetClusterReq
+}
+
+// createAlertmanagerReq defines HTTP request for creating alertmanager
+// swagger:parameters createAlertmanager
+type createAlertmanagerReq struct {
+	cluster.GetClusterReq
+	// in: body
+	// required: true
+	Body apiv2.Alertmanager
+}
+
+func (req *createAlertmanagerReq) validateCreateAlertmanagerReq() error {
+	bodyMap := map[string]interface{}{}
+	if err := yaml.Unmarshal(req.Body.Spec.Config, &bodyMap); err != nil {
+		return fmt.Errorf("can not unmarshal yaml configuration: %w", err)
+	}
+	return nil
+}
+
+// deleteAlertmanagerReq defines HTTP request for deleting alertmanager
+// swagger:parameters deleteAlertmanager
+type deleteAlertmanagerReq struct {
```

Comment:
you can merge `deleteAlertmanagerReq` and `getAlertmanagerReq` into `alertManagerReq` as they are the same, you will save on less decode funcs, and `createAlertmanagerReq` can embed it.

Reply:
Thanks for pointing out this, I put them in different requests because I am not sure if they will have different query parameters in the future.
What do you think? If we embed them now, and in the future, we need to split them, will it affect the user side?

Reply:
It would affect the guys who use our generated client, otherwise the pure API should not be affected. 

But if we expect that they will have different params in the future, lets keep them separated

Reply:
Thanks for explaining this, I would keep them separate for now :)